packages/pluginmanager-extension/src/index.ts

```diff
@@ -0,0 +1,170 @@
+    const trans = translator.load('jupyterlab');
+
+    // Translation strings.
+    const category = trans.__('Plugin Manager');
+    const widgetLabel = trans.__('Advanced Plugin Manager');
+    const refreshPlugins = trans.__('Refresh Plugin List');
+
+    const namespace = 'plugin-manager';
+    const tracker = new WidgetTracker<MainAreaWidget<Plugins>>({
+      namespace: namespace
+    });
+
+    /**
+     * Create a MainAreaWidget for Plugin Manager.
+     */
+    function createWidget(args?: PluginListModel.IConfigurableState) {
+      const model = new PluginListModel({
+        ...args,
+        pluginData: {
+          availablePlugins: app.info.availablePlugins
+        },
+        serverSettings: app.serviceManager.serverSettings,
+        extraLockedPlugins: [
+          PLUGIN_ID,
+          // UI will not proceed beyond splash without `layout` plugin
+          '@jupyterlab/application-extension:layout',
+          // State restoration does not work well without resolver,
+          // can leave user locked out of the plugin manager
+          // (if command palette and menu are disabled too)
+          '@jupyterlab/apputils-extension:resolver'
+        ],
+        translator: translator ?? nullTranslator
+      });
+      const content = new Plugins({
+        model,
+        translator: translator ?? nullTranslator
+      });
+      content.title.label = widgetLabel;
+      content.title.icon = extensionIcon;
+      const main = new MainAreaWidget({ content, reveal: model.ready });
+
+      main.toolbar.addItem(
+        'refresh-plugins',
+        new CommandToolbarButton({
+          id: CommandIDs.refreshPlugins,
+          args: { noLabel: true },
+          commands
+        })
+      );
+
+      return main;
+    }
+
+    // Register commands.
+    commands.addCommand(CommandIDs.open, {
+      label: widgetLabel,
+      execute: args => {
+        const main = createWidget(args);
+        shell.add(main, 'main', { type: 'Plugins' });
+
+        // add to tracker so it can be restored, and update when choices change
+        void tracker.add(main);
+        main.content.model.trackerDataChanged.connect(() => {
+          void tracker.save(main);
+        });
+        return main;
+      }
+    });
+
+    commands.addCommand(CommandIDs.refreshPlugins, {
+      label: args => (args.noLabel ? '' : refreshPlugins),
+      caption: trans.__('Refresh plugins list'),
+      icon: refreshIcon,
+      execute: async () => {
+        return tracker.currentWidget?.content.model
+          .refresh()
+          .catch((reason: Error) => {
+            console.error(
+              `Failed to refresh the available plugins list:\n${reason}`
+            );
+          });
+      }
+    });
+
+    if (palette) {
+      palette.addItem({ command: CommandIDs.open, category });
+    }
+
+    if (menu) {
+      const settingsMenu = menu.settingsMenu;
+      settingsMenu.addItem({ command: CommandIDs.open });
+    }
```

Comment:
After some thought I removed it from the menu. This is not a feature that most users will access frequently, it is on par with JSON setting editor and deserves similar treatment, this is access from command palette and from settings:

![image](https://github.com/jupyterlab/jupyterlab/assets/5832902/ece0528e-98f1-4217-b58e-a6e54b269dc0)

Reply:
It is especially fitting there because many plugins currently implement dummy "Disable" option which pretends to disable them. As extensions start migrating away from it, users will still expect to find the option to disable plugins from Settings widget.